prow/jira/jira.go

```diff
@@ -107,8 +99,29 @@ func NewClient(endpoint string, opts ...Option) (Client, error) {
 		}
 	}
 
-	jiraClient, err := jira.NewClient(retryingClient.StandardClient(), endpoint)
-	return &client{upstream: jiraClient, url: endpoint}, err
+	return jira.NewClient(retryingClient.StandardClient(), endpoint)
+}
+
+func NewClient(endpoint string, opts ...Option) (Client, error) {
+	o := Options{}
+	for _, opt := range opts {
+		opt(&o)
+	}
+
+	log := logrus.WithField("client", "jira")
+	if len(o.LogFields) > 0 {
+		log = log.WithFields(o.LogFields)
+	}
+	retryingClient := retryablehttp.NewClient()
+	usedFlagTransport := &clientUsedTransport{
+		upstream: retryingClient.HTTPClient.Transport,
+	}
+	retryingClient.HTTPClient.Transport = usedFlagTransport
+	retryingClient.Logger = &retryableHTTPLogrusWrapper{log: log}
+
+	jiraClient, err := newJiraClient(endpoint, o, retryingClient)
+	url := jiraClient.GetBaseURL()
```

Comment:
Please return here if err != nil, otherwise this might end up being a nil pointer dereference